src/main/java/ch/njol/skript/lang/util/common/AnyAmount.java

```diff
@@ -0,0 +1,42 @@
+package ch.njol.skript.lang.util.common;
+
+import org.jetbrains.annotations.NotNull;
+
+/**
+ * A provider for anything with a (number) amount/size.
+ * Anything implementing this (or convertible to this) can be used by the {@link ch.njol.skript.expressions.ExprAmount}
+ * property expression.
+ *
+ * @see AnyProvider
+ */
+@FunctionalInterface
+public interface AnyAmount extends AnyProvider {
```

Comment:
Just had an idea, what if we implemented this in CondIsEmpty? Could simply check `amount() == 0`, or it might be worth adding a default method to this interface:
```java
default boolean isEmpty() {
	amount() == 0;
}
```